src/main/java/reactor/netty/resources/NewConnectionProvider.java

```diff
@@ -81,6 +86,41 @@
 		});
 	}
 
+	@Override
+	@SuppressWarnings("unchecked")
+	public <CONF extends TransportConfig, CONN extends Connection> Mono<CONN> acquire(TransportClientConfig<CONF, CONN> config) {
+		return Mono.create(sink -> {
+			SocketAddress remote = config.remoteAddress();
+			if (remote instanceof Supplier) {
+				remote = Objects.requireNonNull(((Supplier<? extends SocketAddress>) remote).get(),
+						"Remote Address supplier returned null");
+			}
+
+			ChannelInitializer<Channel> channelInitializer =
+					config.channelInitializer(new NewConnectionObserver((MonoSink<Connection>) sink, config.connectionObserver()), remote);
+			TransportConnector.connect(config, remote, config.resolver(), channelInitializer).log()
+			                  .subscribe(
+			                          channel -> {
+			                              sink.onCancel(channel::close);
```

Comment:
I wonder if that functionality can be integrated directly in the `TransportConnector`-generated `Mono`?